engine/crates/parser-sdl/src/rules/cors_directive/mod.rs

```diff
@@ -0,0 +1,210 @@
+use engine::registry::CorsConfig;
+use engine_parser::{types::SchemaDefinition, Positioned};
+
+use crate::directive_de::parse_directive;
+
+use super::{
+    directive::Directive,
+    visitor::{Visitor, VisitorContext},
+};
+
+const CORS_DIRECTIVE_NAME: &str = "cors";
+
+#[derive(Debug, Clone, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CorsDirective {
+    max_age: Option<u32>,
+    #[serde(default)]
+    allow_credentials: bool,
+    allowed_headers: Option<AnyOrStringArray>,
+    allowed_methods: Option<AnyOrStringArray>,
```

Comment:
Not going to fight against it, but since we're in control of the valid HTTP methods it seems a bit weird to let users say "no, you can't do a POST actually" (or to allow PUT & DELETE, which we won't accept either).

Reply:
I agree, it was in the "RFC", but I don't think we need these...

Reply:
I would expose only `allowedOrigins`, but that was not in the issue...

Reply:
Discussed with Fredrik, only having `allowedOrigins` and `maxAge` in the settings now.